src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -113,7 +114,15 @@ public void Write(SourceAwareJsonTextWriter writer)
 
             this.EmitMetadata(emitter, program.Metadata);
 
-            this.EmitTypeDefinitionsIfPresent(emitter, program.Types.Concat(ImportClosureInfo.ImportedTypesInClosure));
+            var programTypes = program.Types.Concat(ImportClosureInfo.ImportedTypesInClosure);
+
+            if (programTypes.Any())
+            {
+                TypeAliasLookupVisitor = new ExpressionTypeAliasLookupVisitor();
+                TypeAliasLookupVisitor.Visit(program);
```

Comment:
This would miss any imported types, since they're not part of `program` but just `ImportClosureInfo`. Left a similar comment on the `ExpressionTypeAliasLookupVisitor` class definition, but I would replace with `programTypes.ToImmutableDictionary(t => t.Name)`.